src/mod_filtering.c

```diff
@@ -157,12 +153,8 @@ static void audio_frame_to_sound_texture(struct filtering_ctx *ctx, AVFrame *dst
          * frequency into the imaginary part of the first bin. It also requires
          * the bins allocation to be at least of size (nb_samples + 2) * sizeof(float).
          */
-#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(57, 19, 100)
         ctx->avtx_func(ctx->avtx, bins, bins, sizeof(float));
         bins[1] = bins[nb_samples];
```

Comment:
We don't need to mimic `av_rdft_calc` behavior anymore, so this can be removed, the comment updated, and the magnitude computation below can be simplified (by removing the lowest frequency exception and starting the loop at `0`)

Also, with regards to `rdft_data`, I think it we can now:
- replace `FFTSample` with `float`
- reduce the allocation size to `+1` instead of `+2`

Reply:
Done in a separate commit.